Adding initial test script for pipeline

prietyc123 opened the pull request:
Get pull_requets and push events ci https://github.com/openshift/release/pull/49986 
And consolidate the base test for sample pipeline into the script

mmorhun commented on test/validate-pipeline.sh:
```diff
@@ -0,0 +1,53 @@
+DEBUG_OUTPUT=${DEBUG_OUTPUT:-'/tmp/log.txt'}
+EVENT_TYPE=${EVENT_TYPE:-'push'}
+PIPELINE="${PIPELINE:-docker-build-rhtap}"
+TASK="${TASK:-update-deployment}"
```
The same should apply for `deploy-check` task.
I think we should have a list of tasks to check.

mmorhun commented on test/validate-pipeline.sh:
```diff
@@ -0,0 +1,53 @@
+    fi
+fi
+
+exit "${status:-0}"
```
missing new line

mmorhun commented on test/validate-pipeline.sh:
```diff
@@ -0,0 +1,53 @@
+    fi
+else
+    if oc get taskruns/"$PIPELINE"-"$TASK" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+        echo "Error: Gitops deployment repo not updated on push events"
```
It would show correct message only for `update-deployment` task, but we have `$TASK` as a parameter.

mmorhun reviewed (commented):
Is `/test` common location for such scripts?

prietyc123 commented:
> Is `/test` common location for such scripts?

Yes I have started thinking of that. For all the test related stuff I would like to put into this folder

prietyc123 commented:
@mmorhun addressed your review comment https://github.com/redhat-appstudio/tssc-sample-pipelines/pull/26/commits/385d163bf6ae0fba4c1e69601dd85872d59e57ee

xinredhat commented:
/lgtm

mmorhun commented on test/validate-pipeline.sh:
```diff
@@ -0,0 +1,58 @@
+else
+    for task in "${TASKS[@]}"; do
+        if oc get taskruns/"$PIPELINE"-"$task" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+            echo "Error: Gitops deployment repo not modified on push events by task: $task"
```
You can just have a generic message that mentions the name of the task.

prietyc123 replied:
ok I have change this to https://github.com/redhat-appstudio/tssc-sample-pipelines/pull/26/files#diff-5fee22e7689e2ba53b6dfe4256ee71b76d7a74efc328aa8f58212391998e14a4R45-R52

prietyc123 commented:
/retest

prietyc123 commented:
@mmorhun I think there is something wrong with the pipeline 

```
Error from server (BadRequest): error when creating "/go/src/github.com/redhat-appstudio/tssc-sample-pipelines/pac/pipelines/gitops-pull-request-rhtap.yaml": admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: pipeline task "clone-repository" 
expects workspace with name "git-auth" but none exists in pipeline spec: spec.tasks[0].workspaces[1]
invalid value: pipeline task "clone-repository" expects workspace with name "workspace" 
but none exists in pipeline spec: spec.tasks[0].workspaces[0]
invalid value: pipeline task "gather-deploy-images" expects workspace with name "workspace" 
but none exists in pipeline spec: spec.tasks[1].workspaces[0]
```

jduimovich commented:
> @mmorhun I think there is something wrong with the pipeline
> 
> ```
> Error from server (BadRequest): error when creating "/go/src/github.com/redhat-appstudio/tssc-sample-pipelines/pac/pipelines/gitops-pull-request-rhtap.yaml": admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: pipeline task "clone-repository" 
> expects workspace with name "git-auth" but none exists in pipeline spec: spec.tasks[0].workspaces[1]
> invalid value: pipeline task "clone-repository" expects workspace with name "workspace" 
> but none exists in pipeline spec: spec.tasks[0].workspaces[0]
> invalid value: pipeline task "gather-deploy-images" expects workspace with name "workspace" 
> but none exists in pipeline spec: spec.tasks[1].workspaces[0]
> ```

The pipeline works as run from the pipelinerun[]( https://github.com/redhat-appstudio/tssc-sample-pipelines/blob/main/pac/gitops-repo/gitops-on-pull-request.yaml) as it defines the workspace and auth. 
Perhaps we can change the tests to use tkn-pac as the resolver for the images which the mirrors how it runs in RHTAP via PaC 

We can also fix the pipeline def to allow `oc apply` use

prietyc123 commented:
This should fix the CI issue https://github.com/openshift/release/pull/50089

/test validate-rhtap-sample-pipelines

mmorhun commented:
/approve

prietyc123 commented:
/approve

prietyc123 commented:
/test validate-rhtap-sample-pipelines
